serve/mlc_serve/model/model_common.py

```diff
@@ -89,6 +89,7 @@ def sample_from_logits(
     torch_dtype: torch.dtype,
     torch_dev: str,
     past_decode_tokens: List[List[int]],
+    prompt_masks: List[torch.Tensor],
```

Comment:
My small complaint with previous state was we create torch.tensors for sampling outside of https://github.com/Deelvin/mlc-llm/blob/vc/fix%23214/serve/mlc_serve/model/sampler.py#L103 as this is intended as a singular point of creating/managing them all together. 

As we discussed, having List[List[int]] which can be very large (multiple of 32k) seems to cause the huge regression, guess it is inevitable now. Can we leave a comment about this with `TODO` tag so that we can follow-up?

Reply:
We can, but this TODO in my mind. I plan to start refactoring of it due to it is weak place may be related to another performance reduction. I've explained the reasons earlier.
Due to this I think we discuss design in the other PR

Reply:
Agreed.